client/store/reports/test/utils.js

```diff
@@ -527,3 +528,71 @@ describe( 'getReportTableData()', () => {
 		expect( select( 'wc-api' ).getReportItemsError ).toHaveBeenLastCalledWith( 'coupons', query );
 	} );
 } );
+
+describe( 'timeStampFilterDates', () => {
+	const advancedFilters = {
+		filters: {
+			city: {
+				input: { component: 'Search' },
+			},
+			my_date: {
+				input: { component: 'Date' },
+			},
+		},
+	};
+
+	it( 'should not change activeFilters not using the Date component', () => {
+		const activeFilter = {
+			key: 'name',
+			rule: 'is',
+			value: 'New York',
+		};
+		const timeStampedActiveFilter = timeStampFilterDates( advancedFilters, activeFilter );
+
+		expect( timeStampedActiveFilter ).toMatchObject( activeFilter );
+	} );
+
+	it( 'should append timestamps to activeFilters using the Date component', () => {
+		const activeFilter = {
+			key: 'my_date',
+			rule: 'before',
+			value: '2018-04-04',
+		};
+		const timeStampedActiveFilter = timeStampFilterDates( advancedFilters, activeFilter );
+
+		expect( timeStampedActiveFilter.value ).toHaveLength( 25 );
+	} );
+
+	it( 'should append start of day for "after" rule', () => {
+		const activeFilter = {
+			key: 'my_date',
+			rule: 'after',
+			value: '2018-04-04',
+		};
+		const timeStampedActiveFilter = timeStampFilterDates( advancedFilters, activeFilter );
+		expect( timeStampedActiveFilter.value ).toContain( 'T00:00:00+00:00' );
```

Comment:
Also think this would read a bit better if it explicitly had an equality test for the value here and in the before test below too. But just my .02 :)